src/test/java/org/junit/tests/experimental/results/ResultMatchersTest.java

```diff
@@ -20,4 +29,29 @@ public void hasFailuresDescriptionReflectsInput(int i) {
         assertThat(ResultMatchers.failureCountIs(i).toString(),
                 containsString("" + i));
     }
+
+    @Test
+    public void testFailureCountIsTakingMatcher_givenResultWithOneFailure() {
+        PrintableResult resultWithOneFailure = new PrintableResult(Collections.singletonList(
+                new Failure(Description.createTestDescription("class", "name"), new RuntimeException("failure 1"))));
+
+        assertThat(ResultMatchers.failureCountIs(equalTo(3)).matches(resultWithOneFailure), is(false));
+        assertThat(ResultMatchers.failureCountIs(equalTo(1)).matches(resultWithOneFailure), is(true));
+    }
+
+    @Test
+    public void failureCountIsTakingMatcherShouldHaveGoodDescription() {
+        Matcher<Integer> matcherTestStub = new BaseMatcher<Integer>() {
```

Comment:
I see what you mean.

I understand this may be a bit OT  but if we take one step back, are we sure we need a lower bound?

    failureCountIs(Matcher<? super Integer>)

I think we need quite the opposite: an upper bound, so as to allow any Matcher<Integer> and _below_, more specific; we do not want to allow something generic like Matcher<Object>, do we?

See also https://docs.oracle.com/javase/tutorial/java/generics/wildcardGuidelines.html:

> An "in" variable is defined with an upper bounded wildcard, using the extends keyword.

But since Integer is final, we may just go for no bound at all: `Matcher<Integer>` or at most `Matcher<? extends Comparable<Integer>>`, which happens to be the [return type of greaterThan](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#greaterThan%28T%29).

        ResultMatchers.failureCountIs(allOf(equalTo(2), greaterThan(2)));

The above line compiles with both `Matcher<Integer>` and `Matcher<? extends Comparable<Integer>>`.

Then, again, I understand that `Matcher<? super Integer>` is what hamcrest uses in cases like this (e.g. in [arrayWithSize](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#arrayWithSize%28org.hamcrest.Matcher%29)) , so I'll understand if you prefer to comply.